internal/services/batch/batch_pool_resource.go

```diff
@@ -159,15 +160,22 @@ func resourceBatchPool() *pluginsdk.Resource {
 										ForceNew:     true,
 										ValidateFunc: validation.StringIsNotEmpty,
 									},
+									"identity_reference": {
```

Comment:
if this is always the user assigned identity could we make it
```suggestion
									"user_assigned_identity_id": {
```